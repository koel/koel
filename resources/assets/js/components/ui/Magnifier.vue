<template>
  <span>
    <button title="Zoom out" type="button" @click.prevent="$emit('out')">
      <icon :icon="faSearchMinus"/>
    </button>
    <button title="Zoom in" type="button" @click.prevent="$emit('in')">
      <icon :icon="faSearchPlus"/>
    </button>
  </span>
</template>

<script lang="ts" setup>
import { faSearchMinus, faSearchPlus } from '@fortawesome/free-solid-svg-icons'

const emit = defineEmits<{ (e: 'in' | 'out'): void }>()
</script>

<style lang="scss" scoped>
span {
  display: flex;
  transition: .2s;

  button {
    @include inset-when-pressed();

    background: var(--color-bg-primary);
    border: 1px solid rgba(255, 255, 255, .2);
    opacity: .8;
    color: var(--color-text-primary);
    transition: background .2s;
    padding: .5rem .75rem;

    &:hover {
      opacity: 1;
      background: var(--color-bg-primary);
      color: var(--color-text-primary);
    }

    &:first-child {
      border-radius: 4px 0 0 4px;
      border-right: 0;
    }

    &:last-child {
      border-radius: 0 4px 4px 0;
    }
  }
}
</style>
