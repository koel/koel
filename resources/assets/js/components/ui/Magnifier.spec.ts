import { expect, it } from 'vitest'
import UnitTestCase from '@/__tests__/UnitTestCase'
import { screen } from '@testing-library/vue'
import Magnifier from './Magnifier.vue'

new class extends UnitTestCase {
  protected test () {
    it('renders and functions', async () => {
      const { html, emitted } = this.render(Magnifier)

      await this.user.click(screen.getByRole('button', { name: 'Zoom in' }))
      expect(emitted().in).toBeTruthy()

      await this.user.click(screen.getByRole('button', { name: 'Zoom out' }))
      expect(emitted().out).toBeTruthy()

      expect(html()).toMatchSnapshot()
    })
  }
}
