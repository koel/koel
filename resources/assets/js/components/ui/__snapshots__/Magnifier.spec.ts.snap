// Vitest Snapshot v1

exports[`renders and functions 1`] = `<span data-v-fcc3eddd=""><button title="Zoom out" type="button" data-v-fcc3eddd=""><br data-testid="icon" icon="[object Object]" data-v-fcc3eddd=""></button><button title="Zoom in" type="button" data-v-fcc3eddd=""><br data-testid="icon" icon="[object Object]" data-v-fcc3eddd=""></button></span>`;
